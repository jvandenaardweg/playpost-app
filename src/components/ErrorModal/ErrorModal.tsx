import React from 'react';
import { View, Text } from 'react-native';
import { Button } from 'react-native-elements';

interface Props {
  onPressAction(url: string): void;
  message: string;
  action: string;
}

export const ErrorModal: React.FC<Props> = ({ message, action, onPressAction }) => (
  <View style={{
    backgroundColor: 'white',
    padding: 14,
    borderRadius: 10
  }}>
    <Text>{message}</Text>

    {action === 'login' && (
      <View>
        <Button title="Close" type="outline" onPress={() => onPressAction('readto://Onboarding')} />
        <Button title="Login" onPress={() => onPressAction('readto://Onboarding')} />
      </View>
    )}

    {action === 'playlist' && (
      <View>
        <Button title="Close" type="outline" onPress={() => onPressAction('readto://Onboarding')} />
        <Button title="Login" onPress={() => onPressAction('readto://Onboarding')} />
      </View>
    )}

  </View>
);
