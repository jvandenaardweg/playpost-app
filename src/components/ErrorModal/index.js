import React from 'react';
import { View, Text } from 'react-native';
import { Button } from 'react-native-elements';

export const ErrorModal = (props) => (
  <View style={{
    backgroundColor: 'white',
    padding: 14,
    borderRadius: 10
  }}>
    <Text>{props.message}</Text>

    {props.action === 'login' && (
      <View>
        <Button title="Close" type="outline" onPress={() => props.onPressAction('readto://Onboarding')} />
        <Button title="Login" onPress={() => props.onPressAction('readto://Onboarding')} />
      </View>
    )}

    {props.action === 'playlist' && (
      <View>
        <Button title="Close" type="outline" onPress={() => props.onPressAction('readto://Onboarding')} />
        <Button title="Login" onPress={() => props.onPressAction('readto://Onboarding')} />
      </View>
    )}

  </View>
);
