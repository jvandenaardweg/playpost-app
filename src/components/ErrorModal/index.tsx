import * as React from 'React';
import { View, Text } from 'react-native';
import { Button } from 'react-native-elements';

interface Props {
  onPressAction(url: string): void;
  message: string;
  action: string;
}

export const ErrorModal = (props: Props) => (
  <View style={{
    backgroundColor: 'white',
    padding: 14,
    borderRadius: 10
  }}>
    <Text>{props.message}</Text>

    {props.action === 'login' && (
      <View>
        <Button title="Close" type="outline" onPress={() => props.onPressAction('readto://Onboarding')} />
        <Button title="Login" onPress={() => props.onPressAction('readto://Onboarding')} />
      </View>
    )}

    {props.action === 'playlist' && (
      <View>
        <Button title="Close" type="outline" onPress={() => props.onPressAction('readto://Onboarding')} />
        <Button title="Login" onPress={() => props.onPressAction('readto://Onboarding')} />
      </View>
    )}

  </View>
);
