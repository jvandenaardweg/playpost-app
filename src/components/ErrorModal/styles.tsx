import {
  StyleSheet
} from 'react-native';

import colors from '../../constants/colors';
import layout from '../../constants/layout';
import spacing from '../../constants/spacing';

export default StyleSheet.create({
  container: {
    backgroundColor: 'white',
    padding: spacing.default,
    borderRadius: layout.borderRadius.large
  },
  footer: {
    marginTop: spacing.default
  },
  messageText: {
    color: colors.black
  }
});
