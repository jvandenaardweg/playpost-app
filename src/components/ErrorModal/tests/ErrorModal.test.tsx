import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { ErrorModal } from '../ErrorModal';

describe('ErrorModal', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<ErrorModal onPressAction={() => {}} message="Test" action="Test" />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
