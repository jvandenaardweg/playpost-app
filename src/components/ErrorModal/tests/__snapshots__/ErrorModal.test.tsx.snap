// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal rendering should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 10,
      "padding": 12,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 12,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="ErrorModal-Button-close"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
